src/Controls/src/Core/Handlers/Items/CarouselViewHandler.Windows.cs

```diff
@@ -151,6 +151,31 @@ protected override ItemsViewScrolledEventArgs ComputeVisibleIndexes(ItemsViewScr
 			return args;
 		}
 
+		protected override void UpdateEmptyViewVisibility()
+		{
+			bool isEmpty = (CollectionViewSource?.View?.Count ?? 0) == 0;
+			if (isEmpty)
+			{
+				if (_loopableCollectionView is not null)
+				{
+					_loopableCollectionView.IsLoopingEnabled = false;
+				}
+
+				ListViewBase.ItemTemplate = null;
+			}
+			else
+			{
+				if (_loopableCollectionView is not null)
+				{
+					_loopableCollectionView.IsLoopingEnabled = true;
+				}
+
+				ListViewBase.ItemTemplate = CarouselItemsViewTemplate;
```

Comment:
Only update `ItemTemplate` when the value are actually changing, which reduces unnecessary UI updates and potential visual glitches.

```
var targetTemplate = isEmpty ? null : CarouselItemsViewTemplate;
if (ListViewBase.ItemTemplate != targetTemplate)
{
    ListViewBase.ItemTemplate = targetTemplate;
}
```

Reply:
@jsuarezruiz, I have updated the code section based on your suggestions.